File: C++/3147.cpp
#include <iostream>

int main() {
    int h, e, a, o, w, x;
    std::cin >> h >> e >> a >> o >> w >> x;
    
    if (h + e + a + x >= o + w)
        std::cout << "Middle-earth is safe." << std::endl;
    else
        std::cout << "Sauron has returned." << std::endl;
    
    return 0;
}

File: C#/3147.cs
using System; 
class URI {
    static void Main(string[] args) {
        string[] l = Console.ReadLine().Split();
        int h = int.Parse(l[0]);
        int e = int.Parse(l[1]);
        int a = int.Parse(l[2]);
        int o = int.Parse(l[3]);
        int w = int.Parse(l[4]);
        int x = int.Parse(l[5]);

        if (h + e + a + x >= o + w) {
            Console.WriteLine("Middle-earth is safe.");
        } else {
            Console.WriteLine("Sauron has returned.");
        }
    }
}
